web3_proxy/src/frontend/authorization.rs

```diff
@@ -1187,8 +1234,43 @@ impl Web3ProxyApp {
                                 None
                             };
 
-                        let rpc_key_id =
-                            Some(rpc_key_model.id.try_into().expect("db ids are never 0"));
+                        // Get the user_tier
+                        let user_model = user::Entity::find_by_id(rpc_key_model.user_id)
+                            .one(db_replica.conn())
+                            .await?
+                            .context("user model was not found, but every rpc_key should have a user")?;
+
+                        let balance = balance::Entity::find()
+                            .filter(balance::Column::UserId.eq(user_model.id))
+                            .one(db_replica.conn())
+                            .await?
+                            .context("related balance was not found but every user should have a balance")?
+                            .available_balance;
```

Comment:
with so many legacy users, I don't think this is true. default to 0

```
.await?.map(|x| x.available_balance).unwrap_or_default();
```

Reply:
It turns out i didnt need that code segment, I deleted it. may lead to merge conflicts with #84 but thats ok